### PR TITLE
Implemented a status verification icon

### PR DESCRIPTION
**Pull request by mihaistate03**
Created a verification badge by instantianting it in a React component and its story, which is called in the AppHeader component. Declared a corresponding function in the fetchers file, which fetches the URL of the verification status from the env file.

Signed-off-by: Mihai State <mihai.state@codethink.co.uk>

### CODE REVIEW

**Comment by orowith2os** [comment]
Is there some documentation on how to use this yet? It's definitely an improvement over the standard "THIS WRAPPER IS NOT AFFILIATED WITH [...]" method of marking an app as unofficial, but, like most things in Flatpak, the documentation might not be there, so people won't know how to use it.

**Comment by barthalion** [comment]
@orowith2os It's neither ready nor merged. There will be documentation when it's officially available.

**Comment by ovflowd** [comment]
Fancy stuff I see here!

**Comment by razzeee** [comment]
Just to be clear, this MR will likely merge in time, but that does not mean, that you can verify your apps (yet)

It should be fine to merge, as this is only the frontend part and won't do anything without the backend (and it's the beta page anyway)

File: frontend/src/env.ts [hunk 1]
```diff
@@ -104,3 +104,7 @@ export const IS_PRODUCTION: boolean =
 export const STRIPE_MAX_PAYMENT = 999999.99
 // Flathub enforces a minimum payment amount of $1 to cover fees
 export const FLATHUB_MIN_PAYMENT = 1
+
+export const APP_BASE_URL: string = `${BASE_URI}/app`
```
Comment (razzeee) [inline frontend/src/env.ts:108]: This is wrong, this base does not exist

File: frontend/src/env.ts [hunk 2]
```diff
@@ -104,3 +104,7 @@ export const IS_PRODUCTION: boolean =
...
+
+export const APP_VERIFICATION_STATUS: string = `${APP_BASE_URL}/get_verification_status`
```
Comment (razzeee) [inline frontend/src/env.ts:110]:
```suggestion
export const APP_VERIFICATION_STATUS = (id: string): string => `${BASE_URI}/${id}/get_verification_status`
```

File: frontend/src/components/application/Details.stories.tsx [hunk 3]
```diff
@@ -2,15 +2,14 @@ import React from "react"
 import { ComponentMeta } from "@storybook/react"
 import Details from "./Details"
 import { faker } from "@faker-js/faker"
-import { AppStats } from "src/types/AppStats"
```
Comment (razzeee) [inline frontend/src/components/application/Details.stories.tsx:5]: Why these changes? Maybe revert them?

File: frontend/src/components/application/AppHeader.tsx [hunk 4]
```diff
@@ -41,7 +47,24 @@ export function AppHeader({
           </div>
         )}
       </div>
-
+      <div
+        style={{
```
Comment (razzeee) [inline frontend/src/components/application/AppHeader.tsx:51]: Please don't use styles directly to style, use tailwind please

File: frontend/src/components/application/AppHeader.tsx [hunk 5]
```diff
@@ -1,28 +1,34 @@
...
   app: Appstream
   user: UserState
   installClicked: (e: any) => void
   donateClicked: (e: any) => void
   vendingSetup: VendingSetup
+  status: VerificationStatus
 }) {
   const { t } = useTranslation()
-
+  //const status: VerificationStatus
```
Comment (razzeee) [inline frontend/src/components/application/AppHeader.tsx:31]: Can probably be removed

File: frontend/src/components/application/Details.stories.tsx (hunk 3 above)
Comment (mihaistate03) [reply frontend/src/components/application/Details.stories.tsx:5]: I made them as a consequence of having the "Cannot find module" error for AppStats, Appstream and Summary. I can change them as they were before, but the error would appear again

File: frontend/src/components/application/Details.tsx [hunk 6]
```diff
@@ -28,26 +26,22 @@ import Lightbox from "react-image-lightbox"
...
-
+import Verification from "./Verification"
 interface Props {
   app?: Appstream
   summary?: Summary
   stats: AppStats
   developerApps: Appstream[]
   projectgroupApps: Appstream[]
 }
-
```
[the remark this answers is not in the document]
Comment (mihaistate03) [reply frontend/src/components/application/Details.tsx:45]: I will change them as they were before

File: frontend/src/env.ts (hunk 1 above)
Comment (mihaistate03) [reply frontend/src/env.ts:108]: I will remove it

File: frontend/src/components/application/AppHeader.tsx (hunk 5 above)
Comment (mihaistate03) [reply frontend/src/components/application/AppHeader.tsx:31]: I will do it

File: frontend/src/components/application/Details.stories.tsx (hunk 3 above)
Comment (razzeee) [reply frontend/src/components/application/Details.stories.tsx:5]: Where did you get that error? I don't think I can replicate and CI was fine before too.

Comment (razzeee) [reply frontend/src/components/application/Details.stories.tsx:5]: NVM, figured it out. Ideally this should be in a different MR.

File: frontend/src/components/application/Details.tsx [hunk 7]
```diff
@@ -28,26 +26,22 @@ import Lightbox from "react-image-lightbox"
...
 import { calculateHumanReadableSize } from "../../size"
 import { useUserContext } from "../../context/user-info"
-
 import "react-image-lightbox/style.css" // This only needs to be imported once in your app
 import { useAsync } from "../../hooks/useAsync"
 import { getAppVendingSetup } from "../../asyncs/vending"
-
 import useCollapse from "react-collapsed"
-
+import Verification from "./Verification"
```
Comment (razzeee) [inline frontend/src/components/application/Details.tsx:33]: Unused

Comment (mihaistate03) [reply frontend/src/components/application/Details.tsx:33]: I`ll remove it

File: frontend/src/env.ts [hunk 8]
```diff
@@ -104,3 +104,7 @@ export const IS_PRODUCTION: boolean =
 export const STRIPE_MAX_PAYMENT = 999999.99
 // Flathub enforces a minimum payment amount of $1 to cover fees
 export const FLATHUB_MIN_PAYMENT = 1
+
+export const APP_BASE_URL: string = `${BASE_URI}/app`
+
+export const APP_VERIFICATION_STATUS = (id: string): string => `${BASE_URI}/${id}/get_verification_status`
```
Comment (ElijahPepe) [inline frontend/src/env.ts:110]: This is what's causing tests to fail. Could you add a new line at the beginning of the template literal and the end of the arrow function?

**Review by ElijahPepe — CHANGES_REQUESTED** [review]

File: frontend/src/env.ts (hunk 8 above)
Comment (razzeee) [reply frontend/src/env.ts:110]: That won't fix the tests, it also needs to be called correctly

**Review by ElijahPepe — APPROVED** [review]
